### PR TITLE
Implemented GCNode

### PR DESCRIPTION
**Pull request by Kirsten0604**
Unit tests for PR.

### CODE REVIEW

File: CMakeLists.txt
Comment (btwooton) [inline CMakeLists.txt]: This line does not belong in this CMakeLists.txt, look for the file that defines the `SOURCE_FILES`, the `src/shaka_scheme/system/gc/GCNode.hpp` line should go in that file. Please make this change and push it to your GcNode branch. Any new changes that you push will automatically reflect on this pull request.

File: main.cpp
Comment (btwooton) [inline main.cpp]: Technically, these changes were not a part of the pull request implementing GCNode. This inclusion was part of your sprint 1 task assignment to build familiarity with the source code. While this isn't technically a big deal, as you haven't really broken anything in the source code by creating this closure and binding it in the top level environment, from a pedagogical perspective, changes to main.cpp should be segregated to their own PR, separate from your GCNode implementation. I'll defer to @VermillionAzure as to whether we should have you do a `git checkout main.cpp` from upstream master and then push this version of the file to your origin GcNode branch.

**Review by btwooton — CHANGES_REQUESTED** [review]
I approve this pull request pending the changes highlighted by my comments. Your unit tests confirm that the arrow and unary star operator are functioning such that GCNode will be able to fit the role of the NodePtr alias that is currently in the source code. I would also like @VermillionAzure to review this prior to the merge being approved once your changes have been committed.

File: tst/shaka_scheme/system/gc/unit-GCNode.cpp
Comment (btwooton) [inline tst/shaka_scheme/system/gc/unit-GCNode.cpp]: Please change the name/comment for this test case to `test_arrow_operator`

File: tst/shaka_scheme/system/gc/unit-GCNode.cpp
Comment (btwooton) [inline tst/shaka_scheme/system/gc/unit-GCNode.cpp]: Please complete the name of this unit test case so that it says `test_dereference_operator`. Right now the name is incomplete. Just aids in readability.

**Review by btwooton — APPROVED** [review]
Thanks for making those changes Kirsten. Things look good, and all of the tests appear to be passing. I approve this Pull Request. Well done!

**Comment by btwooton** [comment]
@VermillionAzure Please do a follow up review and merge when you get the chance. It appears that I do not have write access to perform a merge myself for some reason.

**Comment by CinchBlue** [comment]
@btwooton I will give you squash/merge access now. I will perform the code review shortly after that now as well.

**Review by CinchBlue — APPROVED** [review]
The format is slightly off the on the indentation for `GCNode.cpp` and `GCNode.hpp`. The test cases are missing the `// When:` parts. Finally, I think that there should be some copy or move constructors, but I'm not so sure. @btwooton Perhaps you can expand on those semantics later, but I think we need to move faster. Approving.

**Pull request merged**
